core/src/main/clojure/xtdb/log.clj

```diff
@@ -196,17 +196,14 @@
       (.writeObject xtql-op-writer (ClojureForm. op))
       (.endStruct xtql-writer))))
 
-(defn encode-sql-args [^BufferAllocator allocator, query, arg-rows]
-  (let [plan (sql/compile-query query)
-        {:keys [^long param-count]} (meta plan)
-
-        vecs (ArrayList. param-count)]
+(defn encode-sql-args [^BufferAllocator allocator, arg-rows]
+  (let [max-param-count (apply max (map count arg-rows))
```

Comment:
`apply max` will fail here if `arg-rows` is empty - I assume that's handled elsewhere?